### ISSUE TITLE
Ability to fix "broken" pkg-config files for 'external' libraries

### ISSUE THREAD

**Issue by idekker** [build2/build2#59]
**Problem description**

When trying to cross-compile libodb_sqlite using `b` for our target platform (OpenWRT) and I'm running into the issue that `-I/usr/include` is added as a compiler argument. This results in compilation mistakes as it will load standard header files from there which are incompatible with our platform.

After a lot of debugging I figured out that it found a `sqlite3.pc` file in our target staging directory which specifically states that it should add `-I/usr/include` to `Cflags` even though sqlite can be found in `<staging_dir>/usr/include`. Sqlite3 is not build using `b` (we have a requirement that each package is build separately and for sqlite specifically, we're running a platform provided version).

It seems that all pkg-config files generated by our platform's build system do not specifically list the staging directory as path prefix.
I guess the platform developers consciously decided to configure packages this way, which led to the pkg-config files without the staging directory prefix.
Broken pkg-config files may not be that uncommon.

Changing this behaviour for our implementation may have adverse side-effects and brings responsibility of maintaining this change, so this does not have my preference.

**Request**

Ability to fix "broken" pkg-config files, e.g. by specifying a path-prefix for paths mentioned in the `.pc` files (if it is not there) or a regex on the extracted options of the `.pc` files.

**Comment by johan-boule** [comment]
Note that pkg-config is supposed to strip -I/usr/include, unless the PKG_CONFIG_ALLOW_SYSTEM_CFLAGS env var is set. Maybe PKG_CONFIG_SYSROOT_DIR could be useful in your scenario?

**Comment by boris-kolpackov** [comment]
Good point though keep in mind that `build2` doesn't run the `pkg-config` binary instead using `libpkgconf` library (not sure where it's current home; it keeps jumping between VCS providers). Whether this library respects these environment variables is a good question.

**Comment by idekker** [comment]
Looking at https://github.com/pkgconf/pkgconf, the environment variable is read by the cli and passed to the `libpkgconf` library using the `pkgconf_client_set_sysroot_dir` method. Looking at the build2 sources, it does not seem to use this.

**Comment by boris-kolpackov** [comment]
Another functionality that could fit into this overall model is the ability to suppress resolution of `-l<name>` options. Currently we have a hardcoded exclusion list for some well-known `libc` extensions (`-lm`, `-ldl`, etc., as well as PlatformSDK libraries for Windows). Especially for Windows we may need the ability to extend this list on the fly.

**Comment by boris-kolpackov** [comment]
The staged toolchain (and `0.17.0` once released) now provide a mechanism for rewriting sysroot, similar to `PKG_CONFIG_SYSROOT_DIR`. Details are here:

https://build2.org/stage/build2/doc/build2-build-system-manual.xhtml#cc-import-installed-sysroot

It would still be nice at some point to have a more general, regex-based mechanism for fixing up options coming from `.pc` files so I am going to keep this issue open.
